kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -14333,6 +14333,13 @@ public String expandMacros(String url) {
         try {
             ContentResource cr = getResource(ref);
 
+            long contentLength = cr.getContentLength();
+
+            if (contentLength > (10 * 1024 * 1024)) {
+                log.warn("{} is larger than 10MB, returning an empty Optional ...", ref);
+                return Optional.empty();
```

Comment:
Does it make sense to return a message about the content being to large?

Reply:
Done